[IMP]  project_timesheet_time_control: Tell the user no running line was found

yajo opened the pull request:
Imagine this scenario:

1. In tab 1 of the browser, you have opened task 1.
2. In tab 2 of the browser, you have opened task 2.
3. You go to tab 1 and start a timer.
4. Work, work, work...
5. You go to tab 2 and start a timer, stopping that of task 1.
6. Work, work, work...
7. You go to tab 1 and see that timer as running (it is not, but you didn't refresh). You hit stop.

Before this commit, it just seemed like the timer was actually stopped. What did happen behind the scenes is that your view was refreshed, but no timer was touched fortunately.

After this commit, you get a message telling you that there's no timer to stop and that your browser is most likely out of sync. This mimics the behavior previously found when doing the same, but directly in the AAL. Now it's present in projects and tasks too.

@Tecnativa TT19205 https://github.com/OCA/timesheet/pull/280

yajo committed:
[IMP] project_timesheet_time_control: Tell the user no running line was found

Imagine this scenario:

1. In tab 1 of the browser, you have opened task 1.
2. In tab 2 of the browser, you have opened task 2.
3. You go to tab 1 and start a timer.
4. Work, work, work...
5. You go to tab 2 and start a timer, stopping that of task 1.
6. Work, work, work...
7. You go to tab 1 and see that timer as running (it is not, but you didn't refresh). You hit stop.

Before this commit, it just seemed like the timer was actually stopped. What did happen behind the scenes is that your view was refreshed, but no timer was touched fortunately.

After this commit, you get a message telling you that there's no timer to stop and that your browser is most likely out of sync. This mimics the behavior previously found when doing the same, but directly in the AAL. Now it's present in projects and tasks too.

pedrobaeza reviewed (changes requested):
This is what I told you it was happening!

Please include POT + es.po updates as well, and I'll fast-track

pedrobaeza commented:
Also there's no possibility of avoiding code duplication?

yajo commented:
Maybe with a mixin. Do you think it's worth it?

pedrobaeza commented:
Well, I'm seeing a lot of code duplication, also in OCA/timesheet#280, so I think we can explore that possibility

yajo commented:
Can we merge and leave that for later? :roll_eyes:

pedrobaeza commented:
I prefer to do it now.

yajo committed:
[IMP] project_timesheet_time_control: Refactor with mixin

Models related to timesheet time controls now inherit from a mixin that adds most needed logic automatically.

yajo commented:
OK, done.

pedrobaeza reviewed (approved):
Thank you very much. This is much lighter! Fast-tracking it as said.

pedrobaeza commented:
/ocabot merge minor
